ovos_utils/messagebus.py

```diff
@@ -120,13 +127,195 @@ def close(self):
+        """
+        data = data or {}
+        return Message(msg_type, data, context=self.context)
+
+    def reply(self, msg_type, data=None, context=None):
+        """Construct a reply message for a given message
+
+        This will take the same parameters as a message object but use
+        the current message object as a reference.  It will copy the context
+        from the existing message object and add any context passed in to
+        the function.  Check for a destination passed in to the function from
+        the data object and add that to the context as a destination.  If the
+        context has a source then that will be swapped with the destination
+        in the context.  The new message will then have data passed in plus the
+        new context generated.
+
+        Args:
+            msg_type (str): type of message
+            data (dict): data for message
+            context: intended context for new message
+
+        Returns:
+            Message: Message object to be used on the reply to the message
+        """
+        data = deepcopy(data) or {}
+        context = context or {}
+
+        new_context = deepcopy(self.context)
+        for key in context:
+            new_context[key] = context[key]
+        if 'destination' in data:
+            new_context['destination'] = data['destination']
+        if 'source' in new_context and 'destination' in new_context:
+            s = new_context['destination']
+            new_context['destination'] = new_context['source']
+            new_context['source'] = s
+        return Message(msg_type, data, context=new_context)
+
+    def response(self, data=None, context=None):
+        """Construct a response message for the message
+
+        Constructs a reply with the data and appends the expected
+        ".response" to the message
+
+        Args:
+            data (dict): message data
+            context (dict): message context
+        Returns
+            (Message) message with the type modified to match default response
+        """
+        return self.reply(self.msg_type + '.response', data, context)
+
+    def publish(self, msg_type, data, context=None):
+        """
+        Copy the original context and add passed in context.  Delete
+        any target in the new context. Return a new message object with
+        passed in data and new context.  Type remains unchanged.
+
+        Args:
+            msg_type (str): type of message
+            data (dict): date to send with message
+            context: context added to existing context
+
+        Returns:
+            Message: Message object to publish
+        """
+        context = context or {}
+        new_context = self.context.copy()
+        for key in context:
+            new_context[key] = context[key]
+
+        if 'target' in new_context:
+            del new_context['target']
+
+        return Message(msg_type, data, context=new_context)
+
+
+# compat
+FakeMessage = Message
```

Comment:
Should `FakeMessage` be marked for deprecation?

Reply:
no, the point is to use with FakeBus without dragging ovos-bus-client, it was just introduced....

Reply:
Message imported from here should be deprecated, FakeMessage and the hacks for isinstance checks should be deprecated yes

Reply:
Should the defined class then be `FakeMessage` and `Message` defined as a wrapper class with a deprecation warning?